mergesort.rb

```diff
@@ -1,24 +1,58 @@
 def mergesort(a)
     # if the array size is 0|1 then the list is considered sorted, return sorted array
+    if a.length <= 1
+        return a
+    end
+
     # split the list into two halves
+    left, right = split_array(a) # if odd, I put one less element on the left
+
     # merge sort each half
+    left = mergesort(left)
+    right = mergesort(right)
+
     # combine the sorted halves
+    combine(left, right)
 end
 
 def split_array(a)
-    # find the middle index
+    # find the size of the left split
+    size_left = a.length / 2
+
     # split the array in half - take a look at take and drop in the ruby docs, or use ranges
+    left = a.take(size_left)
+    right = a.drop(size_left)
+
     # return left and right halves of array as separate arrays - did you know a method can return multiple arrays? look it up
+    return left, right
 end
 
 # precondition: a and b are sorted
 def combine(a, b)
     # create a results array
-    # counters pointing to the index of the smallest elements in each array
+    results = []
+
     # check that we have elements to compare
+    until a.empty? || b.empty?
+
         # push the smaller element onto the result array
+        if a.first < b.first
+            results << a.shift
+        elsif a.first > b.first
+            results << b.shift
+        else
+            results << a.shift
+            results << b.shift
+        end
+    end
+
     # if there are elements left over in a, move them to result
+    results << a.first unless a.empty?
+
     # if there are elements left over in b, move them to result
+    results << b.first unless b.empty?
+
```

Comment:
I haven't debugged your code too far, but I think part of the issue is that lines 50 and 53 are dropping elements. If a or b have more than one element at this point, you will only take the first one. The others get left out, so your sorted list is missing a few things.